design/automated-provisioning-design.md

```diff
@@ -0,0 +1,147 @@
+It will handle the full provisioning lifecycle for these Gateways: creating a Contour + Envoy instance for new Gateways; continuously ensuring that the Gateway spec and the Contour + Envoy instance’s configuration remain in sync; and deleting the Contour + Envoy instance when the Gateway is deleted.
+
+![drawing](images/gatewayapi-provisioner-overview.png)
+
+Importantly, this provisioning model allows for (a) any number of GatewayClasses to be controlled by the Gateway provisioner; and (b) any number of Gateways per controlled GatewayClass.
+While the exact use cases for many Contour Gateways remain unclear, these one-to-many relationships between controller and GatewayClass, and GatewayClass and Gateway, offer users the full flexibility that the Gateway API spec intends. 
+Additionally, support for this one-to-many relationship is required in order to pass any of the Gateway API conformance tests.
+
+![drawing](images/gatewayapi-resource-relationships.png)
+
+Longer-term, we think there is an opportunity to have multiple Gateways be combined into a single Contour+Envoy instance, by supporting more listeners, collapsing compatible listeners, etc.
+We are not attempting to solve that more complex problem in this design document, but the solution proposed here should take us in the right direction.
+
+Contour itself will continue to handle the vast majority of the Gateway and HTTPRoute/TLSRoute validation and processing.
+It is responsible for programming Envoy to fulfill the Gateway and route specs, and for setting appropriate conditions on resources.
+The provisioner is solely responsible for deploying Contour + Envoy instances, and for setting the “Scheduled” condition on Gateways.
+
+Again, the Gateway provisioner will be an optional component; users who are not interested in Gateway API or dynamic provisioning of Gateways are not required to use it.
+Those users can continue to statically provision their Contour + Envoy instances, for use with either Gateway API or traditional Ingress / HTTPProxy.
+
+There will be two ways to configure Contour for Gateway API support in the static provisioning scenario:
+- **Controller name** - this is the model implemented today, where Contour is configured with a controller name, and it continuously looks for the oldest GatewayClass with that controller, and the oldest Gateway using that GatewayClass. This model is appropriate for users who expect their GatewayClasses and Gateways to come and go, and who want their Contour instance to dynamically pick up the appropriate Gateway as those changes occur.
+- **Gateway name** - Contour can alternately be directly configured with a specific Gateway name, which avoids the multiple levels of indirection of the previous model. This model is appropriate for users who expect their Contour instance to correspond to a single static Gateway; the lifecycle of the Gateway and the lifecycle of the Contour instance are tied together.
```

Comment:
Isn't this what the provisioner will use though? It will directly tie a Contour to a Gateway?

Reply:
Yep, this is how the provisioner will configure Contour. But it can also be done directly by a user in the static provisioning mode. I'll add some clarifying text.